### PR TITLE
Implementing Sort in feature

### LINKED ISSUE #7485: Sorting in features page
**Please describe the desired behavior.**
Describe the solution you'd like.
Currently there is no kind of sorting on features page .It would be benficial if we could sort features based on title or last edited.

**Additional context (optional)**
Feel free to include any applicable: code, screenshots, or gifs.

### PR DESCRIPTION
**Pull request by sssash18**
Fixes #7485  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below
**New Behaviour**

Untitled_ Feb 14, 2020 12_35 AM
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by nstjean** [comment]
Very nice job!! Works for me! I do think we're going to have to write a test for the new function to get the test to pass.

We'll add a new functional test here:
https://github.com/publiclab/plots2/blob/master/test/functional/features_controller_test.rb

Look at the example of testing the update feature, I think you can copy a lot of how it works.
https://github.com/publiclab/plots2/blob/d4f803c1dba59f91c5d8eb6e21b9e2c218b94372/test/functional/features_controller_test.rb#L92-L103

You can run just this page of tests by using the command `rails test test/functional/features_controller_test.rb`

Do you think this is something you can do?

**Comment by sssash18** [comment]
Yes @nstjean I will do it.

**Comment by sssash18** [comment]
@nstjean I will implement the 'sort by title' test but for 'sort by last change', currently in nodes.yml both features have same time as last changed , so should'nt we modify it for the test.Thanks

**Comment by nstjean** [comment]
> @nstjean I will implement the 'sort by title' test but for 'sort by last change', currently in nodes.yml both features have same time as last changed , so should'nt we modify it for the test.Thanks

That's a very good point! Yes, you'll have to change one of them to be earlier - maybe by subtracting some time? Then you'll know for sure what order they should be appearing in.

**Comment by sssash18** [comment]
@nstjean @cesswairimu @SidharthBansal Please review the tests added

File: app/controllers/features_controller.rb [hunk 1]
```diff
@@ -88,4 +90,15 @@ def update
...
+  def sort_feature(features, sorting_type)
+    case sorting_type
+    when 'title'
+      features.sort_by(&:title.downcase)
+    when 'last_edited'
+      features.sort_by(&:changed).reverse!
+    else
+      features
+    end
+  end
```
Comment (cesswairimu) [inline app/controllers/features_controller.rb]: Thanks for working on this...lets move this to the model...actually we had another method for sorting in the node model...we can reuse that...Let me see if we have merged that PR

**Comment by sssash18** [comment]
@cesswairimu can you tell me what to do if you have checked that PR?

**Comment by SidharthBansal** [comment]
I agree with @cesswairimu

**Comment by cesswairimu** [comment]
@sssash18 just saw its your PR https://github.com/publiclab/plots2/pull/7481
We can find a way to reuse the method you had here on this sorting...What do you think?

**Comment by sssash18** [comment]
@cesswairimu It can be done but the number of params for sorting are different for both the cases.

**Comment by cesswairimu** [comment]
Lets have the othe PR merged and I will help in refactoring the method to reuse for both

**Comment by sssash18** [comment]
@cesswairamu I am going to make the suggested changes in the next week, currently just busy with my college exams. Sorry for the delay.

**Comment by cesswairimu** [comment]
No worries, just ping when its ready and all the best in your exams

**Pull request closed without merging**